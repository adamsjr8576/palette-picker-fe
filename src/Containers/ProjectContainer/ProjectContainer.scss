.section-project-container {
  align-items: center;
  justify-content: space-around;
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}

.project-cards-div {
  text-align: center;
  width: 100%;
}