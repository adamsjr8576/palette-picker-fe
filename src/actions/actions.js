export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
})