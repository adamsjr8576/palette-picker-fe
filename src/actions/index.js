export const addNewPalette = currentPalette => ({
  type: 'ADD_PALETTE',
  currentPalette
});

export const updatePaletteLocked = hexCode => ({
  type: 'UPDATE_PALETTE_LOCKED',
  hexCode
});

export const addInitialProjects = projects => ({
  type: 'ADD_INITIAL_PROJECTS',
  projects
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTE_TO_PROJECT',
  palettes
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  project
});

export const deletePaletteFromStore = palette => ({
  type: 'DELETE_PALETTE',
  palette
});

export const deleteProjectFromStore = project => ({
  type: 'DELETE_PROJECT',
  project
});
