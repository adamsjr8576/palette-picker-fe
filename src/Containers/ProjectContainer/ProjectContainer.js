import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { addInitialProjects } from '../../actions/index';
import ProjectCard from '../../Components/ProjectCard/ProjectCard';
import './ProjectContainer.scss';

export const ProjectContainer = ({ addInitialProjects, projects }) => {

  let projectCards

  useEffect(() => {
    fetch(process.env.REACT_APP_BACKEND_URL + '/api/v1/projects') 
      .then(response => response.json())
      .then(data => {
        addInitialProjects(data)
      })
  }, []);

  if(projects.length) {
    projectCards = projects.map(project => {
      return(
        <ProjectCard 
          name={project.name}
          id={project.id}
        />
      )
    })
  }

  return(
    <div>
      <h2>Your Projects</h2>
      {projectCards}
    </div>
  )
}

export const mapStateToProps = state => ({
  projects: state.projects
});

export const mapDispatchToProps = dispatch => ({
  addInitialProjects: projectName => dispatch( addInitialProjects(projectName) )
});

export default connect(mapStateToProps, mapDispatchToProps)(ProjectContainer);