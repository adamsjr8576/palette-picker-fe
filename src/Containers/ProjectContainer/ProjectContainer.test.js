import React from 'react';
import { ProjectContainer, mapDispatchToProps, mapStateToProps } from './ProjectContainer';
import { addInitialProjects } from '../../actions/index';
import { shallow } from 'enzyme';

describe('PaletteGenerator', () => {

  let mockAddInitialProjects = jest.fn();
  let mockProjects = [{name: 'Project1', id: 5}, {name: 'Project2', id: 6}];
  let wrapper;

  describe('PalleteGenerator', () => {

    it('should match the snapshot with an array of projects', () => {
      wrapper = shallow(<ProjectContainer
        addInitialProjects={mockAddInitialProjects}
        projects={mockProjects}
        />)

      expect(wrapper).toMatchSnapshot()
    });

    it('should match the snapshot where projects is an empty array', () => {
      mockProjects = [];
      wrapper = shallow(<ProjectContainer
        addInitialProjects={mockAddInitialProjects}
        projects={mockProjects}
        />)

      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    test('should return an object with the projects data', () => {
      const mockState = {
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}],
        projects: [{name: 'Project1', id: 5}, {name: 'Project2', id: 6}]
      };
      const expected = {
        projects: [{name: 'Project1', id: 5}, {name: 'Project2', id: 6}]
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch with addInitialProjects action ', () => {
      const projects = [{name: 'Project1', id: 5}, {name: 'Project2', id: 6}];
      const mockDispatch = jest.fn();
      const actionToDispatch = addInitialProjects(projects);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addInitialProjects(projects);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });
});
