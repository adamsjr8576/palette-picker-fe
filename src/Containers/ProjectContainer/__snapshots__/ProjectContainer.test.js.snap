// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteGenerator PalleteGenerator should match the snapshot where projects is an empty array 1`] = `
<div
  className="project-cards-div"
>
  <h2
    className="your-projects-h2"
  >
    Your Projects
  </h2>
  <section
    className="section-project-container"
  />
</div>
`;

exports[`PaletteGenerator PalleteGenerator should match the snapshot with an array of projects 1`] = `
<div
  className="project-cards-div"
>
  <h2
    className="your-projects-h2"
  >
    Your Projects
  </h2>
  <section
    className="section-project-container"
  >
    <Connect(ProjectCard)
      id={5}
      key="5"
      name="Project1"
    />
    <Connect(ProjectCard)
      id={6}
      key="6"
      name="Project2"
    />
  </section>
</div>
`;
